ykrt/src/compile/mod.rs

```diff
@@ -60,21 +94,28 @@ pub struct CompiledTrace {
     /// Pointer to heap allocated live AOT values.
     aotvals: SendSyncConstPtr<c_void>,
     /// List of guards containing hotness counts or compiled side traces.
-    guards: Vec<Option<Guard>>,
+    pub guards: Vec<Guard>,
     /// If requested, a temporary file containing the "source code" for the trace, to be shown in
     /// debuggers when stepping over the JITted code.
     ///
     /// (rustc incorrectly identifies this field as dead code. Although it isn't being "used", the
     /// act of storing it is preventing the deletion of the file via its `Drop`)
     #[allow(dead_code)]
     di_tmpfile: Option<NamedTempFile>,
+    /// Reference to the HotLocation, required for side tracing.
+    pub(crate) hl: Weak<Mutex<HotLocation>>,
 }
 
 impl CompiledTrace {
     /// Create a `CompiledTrace` from a pointer to an array containing: the pointer to the compiled
     /// trace, the pointer to the stackmap and the size of the stackmap, and the pointer to the
     /// live AOT values.
```

Comment:
This comment is now out of date with the `hl` argument.

Reply:
Is it? We still need this `hl` for side-tracing. We've just changed it from an `Arc` to a `Weak`.

Reply:
I mean the comment needs to mention `hl`.

Reply:
Fixed in c515a462

Reply:
s/passed in references/references passed in/

Reply:
What I meant is "the references to MT and HotLocation that are passed in as arguments". Your change makes it sound like we are passing stuff into MT and HotLocation which isn't the case here.

Reply:
Fair enough, though "passed in references" doesn't quite parse in English.

Reply:
Better? fb3470abdb